sourcegraph/sourcegraph
# cody: auto signin with app

Sign in user automatically when:
- cody is able to find app token
- app is running
- not signed into another instance when cody starts

it also stores the token from the local file to secret storage automatically when found

## Test plan



See Loom:

https://github.com/sourcegraph/sourcegraph/assets/68532117/d4ad084a-3e68-4c52-809b-1c7b0c97e476

1. Add Cody.experimental.app.connect = true in your vs code user [settings.json](https://code.visualstudio.com/docs/getstarted/settings#_settingsjson)
2. start app and and get token from app
2. create the app.json file manually (mimic the file that will be created by app when it's ready)
3. reload VS Code, you should now be signed in automatically (you will see email verification error which will be removed next week)
